packages/backend-common/src/cache/CacheClient.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { JsonValue } from '@backstage/config';
+import cacheManager from 'cache-manager';
+import { createHash } from 'crypto';
+
+type CacheClientArgs = {
+  client: cacheManager.Cache;
+  pluginId: string;
+};
+
+/**
+ * A pre-configured, storage agnostic cache client suitable for use by
+ * Backstage plugins.
+ */
+export interface CacheClient {
+  /**
+   * Reads data from a cache store for the given key.
+   */
+  get(key: string): Promise<JsonValue>;
+
+  /**
+   * Writes the given data to a cache store, associated with the given key. An
+   * optional TTL may also be provided, otherwise it defaults to the TTL that
+   * was provided when the client was instantiated.
+   */
+  set(key: string, value: JsonValue, ttl?: number): Promise<void>;
+
+  /**
+   * Removes the given key from the cache store.
+   */
+  delete(key: string): Promise<void>;
+}
+
+/**
+ * A simple, concrete implementation of the CacheClient, suitable for almost
+ * all uses in Backstage.
+ */
+export class ConcreteCacheClient implements CacheClient {
+  private readonly client: cacheManager.Cache;
+  private readonly pluginId: string;
+
+  constructor({ client, pluginId }: CacheClientArgs) {
+    this.client = client;
+    this.pluginId = pluginId;
+  }
+
+  async get(key: string): Promise<JsonValue> {
+    const k = this.getNormalizedKey(key);
+    try {
+      const data = (await this.client.get(k)) as string | undefined;
+      return this.unserializeData(data);
+    } catch (_e) {
+      return null;
```

Comment:
Hm so, is this a silent behaviour that we want? At least it should probably return `undefined` to make it look not-present, and maybe have an optional `options: { onError: 'reject' | 'returnEmpty' }` on `get` with some default? Not sure

Reply:
Went with the `onError` pattern at the client-level.  I like the simplicity of not having to worry about catching errors for most cases, but can understand others where error handling is important.